lib/dest/writeContents/index.js

```diff
@@ -36,27 +100,65 @@ function writeContents(writePath, file, cb) {
     cb(err, file);
   }
 
-  function written(err) {
-
-    if (isErrorFatal(err)) {
-      return complete(err);
+  function written(err, fd) {
+    function complete2(err1) {
+      fs.close(fd, function(err2) {
+        complete(err1 || err2);
+      });
     }
 
-    if (!file.stat || typeof file.stat.mode !== 'number' || file.symlink) {
-      return complete();
+    function applyStatFd(err, st) {
+      if (err) {
+        return complete2(err);
+      }
+
+      var mode = getMode(st, file.stat);
+      if(mode == null) {
+        return futimes(fd, st, file.stat, complete2);
+      }
+
+      fs.fchmod(fd, mode, function(error) {
+        if (error) {
+          return complete2(error);
+        }
+
+        futimes(fd, st, file.stat, complete2);
+      });
     }
 
-    fs.stat(writePath, function(err, st) {
+    function applyStatPath(err, st) {
       if (err) {
         return complete(err);
       }
-      var currentMode = (st.mode & parseInt('0777', 8));
-      var expectedMode = (file.stat.mode & parseInt('0777', 8));
-      if (currentMode === expectedMode) {
-        return complete();
+
+      var mode = getMode(st, file.stat);
+      if(mode == null) {
+        return utimes(writePath, st, file.stat, complete);
       }
-      fs.chmod(writePath, expectedMode, complete);
-    });
+
+      fs.chmod(writePath, mode, function(error) {
+        if (error) {
+          return complete(error);
+        }
+
+        utimes(writePath, st, file.stat, complete);
+      });
+    }
+
+
+    if (isErrorFatal(err)) {
+      return complete(err);
+    }
+
+    if (!file.stat) {
+      return complete();
+    }
+
+    if (typeof fd === 'number') {
+      return fs.fstat(fd, applyStatFd);
```

Comment:
I'm using node 0.10 (the furthest back we support).  I did some googling and it seems that it is expected behavior as long as the flag is set as read-only instead of read-write